src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = {};
+  const dateValues = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateParts[fromFormat[i]] = dateValues[i];
+  }
+
+  if (dateParts['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateParts['YY'], 10);
+
+    if (year < 30) {
+      dateParts['YYYY'] = `20${dateParts['YY']}`;
+    } else {
+      dateParts['YYYY'] = `19${dateParts['YY']}`;
+    }
+  }
+
+  if (dateParts['YYYY'] && !dateParts['YY'] && toFormat.includes('YY')) {
+    dateParts['YY'] = dateParts['YYYY'].slice(-2);
+  }
+
+  const formattedDate = toFormat
+    .filter((part) => part !== toSeparator)
+    .map((part) => dateParts[part])
+    .join(toSeparator);
```

Comment:
The logic that constructs the `formattedDate` is concise and functional. However, consider adding a comment above this section to describe what the code is doing for the sake of clarity, especially for less experienced developers who might be reading this code.